website/content/docs/security/acl/auth-methods/kubernetes.mdx

```diff
@@ -158,3 +158,13 @@ use in binding rule selectors and bind name interpolation.
 | `serviceaccount.namespace` | Equal, Not Equal, In, Not In, Matches, Not Matches | yes                 |
 | `serviceaccount.name`      | Equal, Not Equal, In, Not In, Matches, Not Matches | yes                 |
 | `serviceaccount.uid`       | Equal, Not Equal, In, Not In, Matches, Not Matches | yes                 |
+
+## Token Locality
+
+By default, tokens created by the Kubernetes Auth Method will have local scope
+and will therefore only be valid for the datacenter in which they were created.
+For cases requiring global token scope, the primary Consul datacenter must have
+access to all Kubernetes clusters in order for tokens to be validated. This
+property is often prohibitive so using [`-token-locality=global`](/commands/acl/auth-method/create#token-locality)
```

Comment:
I don't think I follow why Consul would need access to all kube clusters. Couldn't it create global tokens from the one cluster used by the primary DC? I guess that assumes that the primary cluster contains all the users, but maybe that's a fine assumption sometimes?

Reply:
@mkeeler could you chime in here? I was basing this aspect on the comment [you made](https://hashicorp.slack.com/archives/C0253EQ5B40/p1629138964222900?thread_ts=1629133907.212800&cid=C0253EQ5B40): 
>  the caveat with global tokens from authmethods is that its not always possible to configure it that way. For example if you are using the k8s auth method and the primary dc doesn’t have access to every one of your k8s clusters. Then you cannot use global tokens there and may have to resort to the anonymous token approach.

Reply:
The k8s auth method regardless of which DC its in and what locality you assign **must** have access to the k8s APIs of the cluster that is issuing the services JWT tokens that will be used for Consul's ACL login flow.

With that comment I was trying to say that you can only rely on just having k8s auth-methods that generate global tokens (in the primary dc) if all the k8s clusters you want to allow logins from can be accessed from the primary DC. 

If your primary dc doesn't have access to all the associated clusters then you must configure the anon token to allow those disconnected k8s cluster to login and get acl tokens.

> Couldn't it create global tokens from the one cluster used by the primary DC?

One k8s cluster cannot validate the JWT tokens generated for services in other k8s clusters. We need a separate auth-method for every cluster which means network/API access to each cluster from wherever the auth method is defined.

Reply:
Thank you for clarifying!

If I understand this correctly, there is nothing preventing someone from setting up a k8s auth method in the primary DC, and configuring that auth method to produce global tokens.

The only requirement is that the JWT they pass in comes from that one k8s cluster configured in the auth method. 

There's no requirement that the consul primary DC have access to all k8s clusters, if the JWTs are only ever coming from 1 cluster.

Does that sound right?

Would it be more correct to focus on the requirement of JWTs to Auth methods to network access?

My concern is that statements that claim "Consul must have access to all k8s clusters" is misleading. That's only true if you want a JWT to come from any cluster. If you have setup all service accounts in a single k8s cluster, then you only need an auth method for that one. But you do need to make sure the JWT all come from that one.

Reply:
ok, I _think_ I have addressed this in some meaningful way with the latest revision. Please let me know what you think 🙏🏼